src/gc.c

```diff
@@ -504,7 +509,7 @@ void gc_mark_alloc(GarbageCollector* gc, void* ptr)
         /* Iterate over allocation contents and mark them as well */
         LOG_DEBUG("Checking allocation (ptr=%p, size=%lu) contents", ptr, alloc->size);
```

Comment:
if ( alloc->size < PTRSIZE ) {
        return;
    }

Reply:
That is covered in the `char*` pointer arithmetic and the condition of the `for` loop, since `p <= (char*) alloc->ptr + alloc->size - PTRSIZE` will not eval to `true` for `alloc->size < PTRSIZE`.

Am I missing something?

Reply:
At least the `for` loop run once. And the `If` Statement can avoid that.